contracts/child/StateReceiver.sol

```diff
@@ -159,6 +159,9 @@ contract StateReceiver is System {
             abi.encodeWithSignature("onStateReceive(uint256,address,bytes)", obj.id, obj.sender, obj.data)
         );
 
+        // if state sync fails, revert flag
+        if (!success) processedStateSyncs[obj.id] = false;
```

Comment:
Let's not emit the event in this case.

Reply:
I think it's necessary to emit the event, so that if a batch execute finishes, a user can still fetch whether their state sync was successful or not